src/planscape/forsys/views.py

```diff
@@ -189,21 +217,47 @@ def run_forsys_generate_project_areas_for_a_single_scenario(
 def generate_project_areas_for_a_single_scenario(
         request: HttpRequest) -> HttpResponse:
     try:
+        pr = cProfile.Profile()
+        if settings.DEBUG:
+            _set_up_cprofiler(pr)
+
         params = ForsysGenerationRequestParams(request)
         headers = ForsysInputHeaders(params.priorities)
         forsys_input = ForsysGenerationInput(params, headers)
         forsys_output = run_forsys_generate_project_areas_for_a_single_scenario(
             forsys_input.forsys_input, headers.FORSYS_PROJECT_ID_HEADER,
             headers.FORSYS_STAND_ID_HEADER, headers.FORSYS_AREA_HEADER,
             headers.FORSYS_COST_HEADER, headers.FORSYS_GEO_WKT_HEADER,
-            headers.priority_headers, params.priority_weights)
+            headers.priority_headers, params.priority_weights,
+            "test_scenario" if settings.DEBUG else None,
+            datetime.now().astimezone(
+                timezone('US/Pacific')
+            ).strftime("%Y%m%d-%H-%M") if settings.DEBUG else None)
 
         response = {}
         response['forsys'] = {}
         response['forsys']['input'] = forsys_input.forsys_input
         response['forsys']['output'] = forsys_output.scenario
 
+        if settings.DEBUG:
+            _tear_down_cprofiler(pr, 'output/cprofiler.log')
+
         return JsonResponse(response)
     except Exception as e:
         logger.error('project area generation error: ' + str(e))
         return HttpResponseBadRequest("Ill-formed request: " + str(e))
+
+
+# This enables memory profiling of generate_project_areas_for_a_single_scenario
```

Comment:
should this be in a function or method?

Reply:
nope.

profile is normally used as a decorator:
```
@profile
def generate_project_areas_for_a_single_scenario(...) { ... }
```

but I don't want to enable it in production, and another way to express the decorator is
```
def generate_project_areas_for_a_single_scenario(...) { ... }
generate_project_areas_for_a_single_scenario = profile()(
        generate_project_areas_for_a_single_scenario)
```